### PR TITLE
Fix SnowflakeConnector connection URL

### PR DESCRIPTION
**Pull request by sfc-gh-chu**
# Description

We need a valid database url to launch the dashboard. This PR fixes the snowflake connector DB url (unless a snowpark session is given in which case we can't get the user credentials from the session afaik).

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

----

> [!IMPORTANT]
> Fixes `SnowflakeConnector` to ensure valid database URL construction and handle parameter mismatches and warnings with snowpark sessions.
> 
>   - **Behavior**:
>     - Fixes `SnowflakeConnector` to construct a valid database URL even when a `snowpark_session` is provided.
>     - Adds warning if `password` is not provided with `snowpark_session`.
>     - Checks for mismatched connection parameters between `snowpark_session` and provided arguments.
>   - **Functions**:
>     - Updates `__init__` and `_init_with_snowpark_session` in `connector.py` to handle connection parameters and warnings.
>   - **Misc**:
>     - Adds `connection_parameters` argument to `_init_with_snowpark_session`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=truera%2Ftrulens&utm_source=github&utm_medium=referral)<sup> for 774850ef74f80b790356c4ec8adb3aff5811d5ba. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: src/connectors/snowflake/trulens/connectors/snowflake/connector.py [hunk 1]
```diff
@@ -126,26 +124,22 @@ def _init_with_snowpark_session(
...
 
         database_url = URL(
-            account=snowpark_session.get_current_account(),
-            user=snowpark_session.get_current_user(),
-            password="password",
-            database=snowpark_session.get_current_database(),
-            schema=snowpark_session.get_current_schema(),
-            warehouse=snowpark_session.get_current_warehouse(),
-            role=snowpark_session.get_current_role(),
+            **connection_parameters,
```
Comment (sfc-gh-dkurokawa) [inline src/connectors/snowflake/trulens/connectors/snowflake/connector.py:142]: won't we still have the same problem with the dashboard if ppl use a `snowflake_session` to connect?

Comment (sfc-gh-chu) [reply src/connectors/snowflake/trulens/connectors/snowflake/connector.py:142]: yea, I'll add a warning in that case

Comment (sfc-gh-chu) [reply src/connectors/snowflake/trulens/connectors/snowflake/connector.py:142]: we'd need it for server-side evals. In the warehouse env, we only have a snowpark session

Comment (sfc-gh-dkurokawa) [reply src/connectors/snowflake/trulens/connectors/snowflake/connector.py:153]: FYI: this isn't technically correct as you can use the DB url to do it. That's how the stored proc works right now as we don't even use `SnowflakeConnector` and just use the DB url. Arguably, this should be changed to use the `SnowflakeConnector` though. I'm not sure as I haven't thought what's best.

**Review by sfc-gh-dkurokawa — APPROVED** [review]

**Pull request merged**